controller/linodecluster_controller.go

```diff
@@ -186,11 +248,52 @@ func (r *LinodeClusterReconciler) reconcileCreate(ctx context.Context, logger lo
 		Port: int32(linodeNBConfig.Port),
 	}
 
+	// Handle firewalls
+	firewall, err := services.HandleFirewall(ctx, clusterScope.ControlPlaneFirewall, clusterScope.LinodeClient, logger)
```

Comment:
IIUC, both the LinodeClusterController and the LinodeFirewallController both call `services.HandleFirewall` and are both reconciling the Cloud Firewall independently. Could this introduce any possible conflicts?